WordPress/gutenberg
# Components: update snackbar to use framer motion instead of react spring

This PR experiments with using framer motion instead of react spring for the snackbar component. Since Framer Motion is ~30kb gzipped I also ~externalized it~ exported this from `@wordpress/components` once to avoid bloating each package that it's used in.

One benefit here is that framer motion supports `height: auto` which allows us to remove ref and component state management. 🤔 ~There's quite a bundle size jump here to consider for a single component, so I may explore trying to whittle down.~

Results overall, are are encouraging. I'd like to see if we can update the Persistent List View to use drag and add animations for the drawers opening and closing next.

| Size Impacts |
|----|
| <img width="602" alt="Screen Shot 2021-08-04 at 9 54 40 AM" src="https://user-images.githubusercontent.com/1270189/128222441-287b2cd3-4536-4add-94a0-a8094f46cfb2.png"> |

#### After - Motion:

https://user-images.githubusercontent.com/1270189/127399049-aa5a91fe-88e9-4783-857e-7531e9adcf8b.mp4

#### After - Reduce Motion

https://user-images.githubusercontent.com/1270189/127397712-3a733583-5e04-485c-bf2f-791e01950e4d.mp4

#### Before - Motion

https://user-images.githubusercontent.com/1270189/127397745-b6b3c29f-5952-4e1a-87c4-cc27564ad7f7.mp4

#### Before - Reduce Motion

https://user-images.githubusercontent.com/1270189/127397786-38325ff9-563c-4446-a2a7-a0b528d19b80.mp4
 

### Testing Instructions

In console type:
```
wp.data.dispatch('core/notices').createNotice(
		'info',
		'Post published.',
		{
			type: 'snackbar',
			actions: [
				{
					onClick: () => {},
					label: 'View post'
				}
			]
		}
	);
```
Or perform an action like publishing or updating a page.

Verify that folks are happy with how framer motion is being exported.

| Verify that there are no regressions in the preview panel |
|-----|
| <img width="245" alt="Screen Shot 2021-08-03 at 12 10 13 PM" src="https://user-images.githubusercontent.com/1270189/128072345-28934d99-042b-4ca7-ada5-fcd5eda4510c.png"> |
